logos/logos-workernode/logos_worker_node/model_cache.py

```
@@ -372,13 +384,161 @@ async def cache_models_by_priority(self, models: list[str]) -> dict[str, str]:
+            logger.info("RAM cache worker cancelled (queue=%s, caching=%s)", list(self._cache_queue), self._caching_now)
+            raise
+
+    async def stop_background_caching(self) -> None:
+        """Cancel the background worker task. Call from app shutdown."""
+        if self._caching_task is None or self._caching_task.done():
+            return
+        self._caching_task.cancel()
+        try:
+            await self._caching_task
+        except (asyncio.CancelledError, Exception):
+            pass
```

Ruff (0.15.15)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)